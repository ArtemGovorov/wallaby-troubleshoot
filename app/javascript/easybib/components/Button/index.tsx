import styled, { css } from 'styled-components'

interface ButtonProps {
  inline?: boolean
  primary?: boolean
  newPrimary?: boolean
  lightGray?: boolean
  tertiary?: boolean
  blue?: boolean
}

const Button = styled.a<ButtonProps>`
  padding: 13px 27px;
  border-width: 2px;
  border-style: solid;
  border-color: transparent;
  font-size: ${props => props.theme.font.text.size.md};
  display: ${props => (props.inline ? 'inline-block' : 'block')};
  font-weight: normal;
  text-align: center;
  vertical-align: middle;
  touch-action: manipulation;
  cursor: pointer;
  background-image: none;
  white-space: nowrap;
  line-height: 1.42857143;
  border-radius: 3px;
  text-decoration: none;
  user-select: none;
  &:hover {
  }
  &:active {
    box-shadow: inset 0 3px 5px rgba(0, 0, 0, 0.125);
  }
  &:active,
  &:focus {
    outline: none;
    text-decoration: none;
  }
  &:disabled,
  &.disabled {
    opacity: 0.3;
  }
  &:focus {
    border-color: ${props => props.theme.colors.blue};
  }

  ${props =>
    props.primary &&
    css`
      border-color: ${props.theme.colors.orange};
      background-color: ${props.theme.colors.white};
      width: auto;
      height: 50px;
      text-decoration: none;
      color: ${props.theme.colors.orange};
      font-size: ${props.theme.font.text.size.md};
      border-radius: 3px;
      font-weight: 400;
      min-width: auto;
      &:hover {
        color: ${props.theme.colors.orangeDark};
        border-color: ${props.theme.colors.orangeDark};
        &:active {
          color: ${props.theme.colors.orangeDarker};
          border-color: ${props.theme.colors.orangeDarker};
        }
      }
      &:focus {
        color: ${props.theme.colors.orange};
        border-color: ${props.theme.colors.blue};
      }
    `};

  ${props =>
    props.newPrimary &&
    css`
      color: ${props.theme.colors.white};
      background: ${props.theme.colors.orange};
      &:hover,
      &:active,
      &:focus {
        color: ${props.theme.colors.white};
      }
    `};

  ${props =>
    props.lightGray &&
    css`
      color: ${props.theme.colors.greyDark};
      background: ${props.theme.colors.greyEEE};
      &:hover,
      &:active,
      &:focus {
        color: ${props.theme.colors.black};
      }
      &:hover {
        background: ${props.theme.colors.greyLight};
        &:active {
          background: ${props.theme.colors.grey};
        }
      }
    `};

  ${props =>
    props.blue &&
    css`
      border-color: ${props.theme.colors.blue};
      background-color: ${props.theme.colors.white};
      width: auto;
      height: 33px;
      padding: 6px 12px;
      border-width: 1px;
      text-decoration: none;
      color: ${props.theme.colors.blue};
      font-size: ${props.theme.font.text.size.md};
      border-radius: 3px;
      font-weight: 400;
      min-width: auto;
      &:hover {
        color: ${props.theme.colors.blue};
        border-color: ${props.theme.colors.blue};
        &:active {
          color: ${props.theme.colors.blue};
          border-color: ${props.theme.colors.blue};
        }
      }
      &:focus {
        color: ${props.theme.colors.blue};
        border-color: ${props.theme.colors.blue};
      }
    `};

  ${props =>
    props.tertiary &&
    css`
      color: ${props.theme.colors.black};
      background: ${props.theme.colors.greylighter};
      &:hover,
      &:active,
      &:focus {
        color: ${props.theme.colors.black};
      }
      &:hover {
        background: ${props.theme.colors.greyLight};
        &:active {
          background: ${props.theme.colors.grey};
        }
      }
    `};
`

export default Button
