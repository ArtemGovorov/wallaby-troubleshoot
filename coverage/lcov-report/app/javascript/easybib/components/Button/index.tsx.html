<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for app/javascript/easybib/components/Button/index.tsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../../index.html">All files</a> / <a href="index.html">app/javascript/easybib/components/Button</a> index.tsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>9/9</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">45.83% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>11/24</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>8/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>9/9</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122
123
124
125
126
127
128
129
130
131
132
133
134
135
136
137
138
139
140
141
142
143
144
145
146
147
148
149
150
151
152
153
154</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import styled, { css } from 'styled-components'
&nbsp;
interface ButtonProps {
  inline?: boolean
  primary?: boolean
  newPrimary?: boolean
  lightGray?: boolean
  tertiary?: boolean
  blue?: boolean
}
&nbsp;
const Button = styled.a&lt;ButtonProps&gt;`
  padding: 13px 27px;
  border-width: 2px;
  border-style: solid;
  border-color: transparent;
  font-size: ${props =&gt; props.theme.font.text.size.md};
  display: ${props =&gt; (props.inline ? <span class="branch-0 cbranch-no" title="branch not covered" >'inline-block' </span>: 'block')};
  font-weight: normal;
  text-align: center;
  vertical-align: middle;
  touch-action: manipulation;
  cursor: pointer;
  background-image: none;
  white-space: nowrap;
  line-height: 1.42857143;
  border-radius: 3px;
  text-decoration: none;
  user-select: none;
  &amp;:hover {
  }
  &amp;:active {
    box-shadow: inset 0 3px 5px rgba(0, 0, 0, 0.125);
  }
  &amp;:active,
  &amp;:focus {
    outline: none;
    text-decoration: none;
  }
  &amp;:disabled,
  &amp;.disabled {
    opacity: 0.3;
  }
  &amp;:focus {
    border-color: ${props =&gt; props.theme.colors.blue};
  }
&nbsp;
  ${props =&gt;
    props.primary &amp;&amp;
    css`
      border-color: ${props.theme.colors.orange};
      background-color: ${props.theme.colors.white};
      width: auto;
      height: 50px;
      text-decoration: none;
      color: ${props.theme.colors.orange};
      font-size: ${props.theme.font.text.size.md};
      border-radius: 3px;
      font-weight: 400;
      min-width: auto;
      &amp;:hover {
        color: ${props.theme.colors.orangeDark};
        border-color: ${props.theme.colors.orangeDark};
        &amp;:active {
          color: ${props.theme.colors.orangeDarker};
          border-color: ${props.theme.colors.orangeDarker};
        }
      }
      &amp;:focus {
        color: ${props.theme.colors.orange};
        border-color: ${props.theme.colors.blue};
      }
    `};
&nbsp;
  ${props =&gt;
    props.newPrimary &amp;&amp;
<span class="branch-1 cbranch-no" title="branch not covered" >    css`</span>
      color: ${props.theme.colors.white};
      background: ${props.theme.colors.orange};
      &amp;:hover,
      &amp;:active,
      &amp;:focus {
        color: ${props.theme.colors.white};
      }
    `};
&nbsp;
  ${props =&gt;
    props.lightGray &amp;&amp;
<span class="branch-1 cbranch-no" title="branch not covered" >    css`</span>
      color: ${props.theme.colors.greyDark};
      background: ${props.theme.colors.greyEEE};
      &amp;:hover,
      &amp;:active,
      &amp;:focus {
        color: ${props.theme.colors.black};
      }
      &amp;:hover {
        background: ${props.theme.colors.greyLight};
        &amp;:active {
          background: ${props.theme.colors.grey};
        }
      }
    `};
&nbsp;
  ${props =&gt;
    props.blue &amp;&amp;
<span class="branch-1 cbranch-no" title="branch not covered" >    css`</span>
      border-color: ${props.theme.colors.blue};
      background-color: ${props.theme.colors.white};
      width: auto;
      height: 33px;
      padding: 6px 12px;
      border-width: 1px;
      text-decoration: none;
      color: ${props.theme.colors.blue};
      font-size: ${props.theme.font.text.size.md};
      border-radius: 3px;
      font-weight: 400;
      min-width: auto;
      &amp;:hover {
        color: ${props.theme.colors.blue};
        border-color: ${props.theme.colors.blue};
        &amp;:active {
          color: ${props.theme.colors.blue};
          border-color: ${props.theme.colors.blue};
        }
      }
      &amp;:focus {
        color: ${props.theme.colors.blue};
        border-color: ${props.theme.colors.blue};
      }
    `};
&nbsp;
  ${props =&gt;
    props.tertiary &amp;&amp;
<span class="branch-1 cbranch-no" title="branch not covered" >    css`</span>
      color: ${props.theme.colors.black};
      background: ${props.theme.colors.greylighter};
      &amp;:hover,
      &amp;:active,
      &amp;:focus {
        color: ${props.theme.colors.black};
      }
      &amp;:hover {
        background: ${props.theme.colors.greyLight};
        &amp;:active {
          background: ${props.theme.colors.grey};
        }
      }
    `};
`
&nbsp;
export default Button
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Oct 27 2018 01:18:42 GMT-0400 (EDT)
</div>
</div>
<script src="../../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../../sorter.js"></script>
</body>
</html>
