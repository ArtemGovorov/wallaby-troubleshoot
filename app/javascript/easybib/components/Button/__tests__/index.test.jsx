import React from 'react'
import shallowWithTheme from 'easybib/utils/shallowTheme'
import Button from '../index'

describe('<Button />', () => {
  it('should render without blowing up', () => {
    const renderedComponent = shallowWithTheme(<Button primary>Button</Button>)
    expect(renderedComponent).toMatchSnapshot()
  })
})
