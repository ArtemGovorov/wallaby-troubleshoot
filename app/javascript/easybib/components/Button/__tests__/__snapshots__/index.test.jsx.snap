// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render without blowing up 1`] = `
.c0 {
  padding: 13px 27px;
  border-width: 2px;
  border-style: solid;
  border-color: transparent;
  font-size: 14px;
  display: block;
  font-weight: normal;
  text-align: center;
  vertical-align: middle;
  touch-action: manipulation;
  cursor: pointer;
  background-image: none;
  white-space: nowrap;
  line-height: 1.42857143;
  border-radius: 3px;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border-color: #ff5313;
  background-color: #fff;
  width: auto;
  height: 50px;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #ff5313;
  font-size: 14px;
  border-radius: 3px;
  font-weight: 400;
  min-width: auto;
}

.c0:active {
  box-shadow: inset 0 3px 5px rgba(0,0,0,0.125);
}

.c0:active,
.c0:focus {
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:disabled,
.c0.disabled {
  opacity: 0.3;
}

.c0:focus {
  border-color: #0090ff;
}

.c0:hover {
  color: #ec4203;
  border-color: #ec4203;
}

.c0:hover:active {
  color: #d23900;
  border-color: #d23900;
}

.c0:focus {
  color: #ff5313;
  border-color: #0090ff;
}

<a
  className="c0"
>
  Button
</a>
`;
